### PR TITLE
feat(viewer): recognise URLs in text value and convert into link (DSP-1595)

### PR DESCRIPTION
**Pull request by kilchenmann**
resolves DSP-1595

s. comment: https://dasch.myjetbrains.com/youtrack/issue/DSP-1595#focus=Comments-4-3243.0-0

### CODE REVIEW

**Comment by kilchenmann** [comment]
This is how it works:

https://user-images.githubusercontent.com/4253580/125464544-e60de2f4-125e-429b-a12d-ee87d47909f8.mov

File: projects/dsp-ui/src/lib/action/pipes/string-transformation/linkify.pipe.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+})
+export class LinkifyPipe implements PipeTransform {
+
+    // constructor(private _domSanitizer: DomSanitizer) { }
+
+    transform(value: string): string {
+        let stylizedText: string = '';
+        if (value && value.length > 0) {
+            for (let str of value.split(' ')) {
+                // if string/url ends with a full stop '.' or column ':' the pipe will not recognize the url
```
Comment (mdelez) [inline projects/dsp-ui/src/lib/action/pipes/string-transformation/linkify.pipe.ts:17]: This comment should be updated to align with line 19. Also, ':' is called a colon in english :)

File: projects/dsp-ui/src/lib/action/pipes/string-transformation/linkify.pipe.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+/**
+ * This pipe analyses a string and converts any url into a href tags
+ */
+@Pipe({
+    name: 'dspLinkify'
+})
+export class LinkifyPipe implements PipeTransform {
+
+    // constructor(private _domSanitizer: DomSanitizer) { }
```
Comment (mdelez) [inline projects/dsp-ui/src/lib/action/pipes/string-transformation/linkify.pipe.ts:11]: can this be removed?

File: projects/dsp-ui/src/lib/action/pipes/string-transformation/linkify.pipe.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+/**
+ * This pipe analyses a string and converts any url into a href tags
```
Comment (mdelez) [inline projects/dsp-ui/src/lib/action/pipes/string-transformation/linkify.pipe.ts:4]: tags -> tag

**Review by mdelez — CHANGES_REQUESTED** [review]

File: projects/dsp-ui/src/lib/action/pipes/string-transformation/linkify.pipe.ts (hunk 1 above)
Comment (kilchenmann) [reply projects/dsp-ui/src/lib/action/pipes/string-transformation/linkify.pipe.ts:17]: oh yes... colon, not column 😉 
fixed in 29c25eb

File: projects/dsp-ui/src/lib/action/pipes/string-transformation/linkify.pipe.ts (hunk 2 above)
Comment (kilchenmann) [reply projects/dsp-ui/src/lib/action/pipes/string-transformation/linkify.pipe.ts:11]: removed in 29c25eb

File: projects/dsp-ui/src/lib/action/pipes/string-transformation/linkify.pipe.ts (hunk 3 above)
Comment (kilchenmann) [reply projects/dsp-ui/src/lib/action/pipes/string-transformation/linkify.pipe.ts:4]: fixed in 29c25eb

File: projects/dsp-ui/src/lib/action/pipes/string-transformation/linkify.pipe.ts (hunk 1 above)
Comment (kilchenmann) [reply projects/dsp-ui/src/lib/action/pipes/string-transformation/linkify.pipe.ts]: and in ab088df

**Review by mdelez — APPROVED** [review]

**Pull request merged**